src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -6251,5 +6251,191 @@ .maxstack  2
+            constantFolding("nuint", "const nuint A = 4294967295; const int B = 1;", "A >> B", "2147483647");
+            constantFolding("nuint", "const nuint A = 1; const int B = 31;", "A >> B", "0");
+            constantFolding("nuint", "const nuint A = 1; const int B = 32;", "A >> B", "1");
+
+            constantFolding("nint", "const nint A = -2147483648; const nint B = 0;", "A & B", "0");
+            constantFolding("nint", "const nint A = -2147483648; const nint B = -1;", "A & B", "-2147483648");
+            constantFolding("nint", "const nint A = -2147483648; const nint B = 2147483647;", "A & B", "0");
+            constantFolding("nuint", "const nuint A = 0; const nuint B = 4294967295;", "A & B", "0");
+            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 4294967295;", "A & B", "2147483647");
+            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 2147483648;", "A & B", "0");
+
+            constantFolding("nint", "const nint A = -2147483648; const nint B = 0;", "A | B", "-2147483648");
+            constantFolding("nint", "const nint A = -2147483648; const nint B = -1;", "A | B", "-1");
+            constantFolding("nint", "const nint A = 2147483647; const nint B = 2147483647;", "A | B", "2147483647");
+            constantFolding("nuint", "const nuint A = 0; const nuint B = 4294967295;", "A | B", "4294967295");
+            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 2147483647;", "A | B", "2147483647");
+            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 2147483648;", "A | B", "4294967295");
+
+            constantFolding("nint", "const nint A = -2147483648; const nint B = 0;", "A ^ B", "-2147483648");
+            constantFolding("nint", "const nint A = -2147483648; const nint B = -1;", "A ^ B", "2147483647");
+            constantFolding("nint", "const nint A = 2147483647; const nint B = 2147483647;", "A ^ B", "0");
+            constantFolding("nuint", "const nuint A = 0; const nuint B = 4294967295;", "A ^ B", "4294967295");
+            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 2147483647;", "A ^ B", "0");
+            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 2147483648;", "A ^ B", "4294967295");
+
+            void constantFolding(string opType, string declarations, string expr, string expectedResult, DiagnosticDescription diagnostic = null)
+            {
+                var diagnostics = (diagnostic is null) ?
+                    (expectedResult is null) ?
+                        new[] { Diagnostic(ErrorCode.ERR_CheckedOverflow, expr) } :
+                        Array.Empty<DiagnosticDescription>() :
+                    new[] { diagnostic };
+                verify(opType, declarations, expr, expectedResult, diagnostics);
+                //verify(opType, declarations, $"checked ({expr})", expectedResult, diagnostics);
```

Comment:
>// [](start = 16, length = 2)

Not sure why you commented this line.

Reply:
Reverted, thanks.

---
In reply to: [387831558](https://github.com/dotnet/roslyn/pull/42143#discussion_r387831558) [](ancestors = 387831558)